cmd/minikube/cmd/start.go

```diff
@@ -668,16 +668,29 @@ func validateSpecifiedDriver(existing *config.ClusterConfig) {
 		return
 	}
 
+	out.WarningT("Start deleting cluster {{.name}} with delete-on-failure flag set", out.V{"name": existing.Name})
+	if viper.GetBool(deleteOnFailure) {
+		// Start failed, delete the cluster
+		profile, err := config.LoadProfile(existing.Name)
+		if err != nil {
+			out.ErrT(style.Meh, `"{{.name}}" profile does not exist, trying anyways.`, out.V{"name": existing.Name})
+		}
+
+		err = deleteProfile(profile)
```

Comment:
verify if this deletes the "Container" too (or only the config)

Reply:
Yes, it does delete the container and config. Updated the description of pr

Reply:
```
🔥  Deleting "minikube" in docker ...
🔥  Deleting container "minikube" ...
🔥  Removing /usr/local/google/home/zyanshu/.minikube/machines/minikube ...
💀  Removed all traces of the "minikube" cluster.
```